### PR TITLE
feat: add CT support for angular 18

### LINKED ISSUE #29309: Support Angular 18 for Component Testing
### What would you like?

Angular 18 is to be released on May 20, 2024 as shown in their release schedule: https://angular.dev/reference/releases#release-schedule

Cypress should support Angular 18.

### Why is this needed?

_No response_

### Other

_No response_

### PR DESCRIPTION
**Pull request by AtofStryker**
- Closes #29309

### Additional details

This PR adds support for Angular 18 CT which was released last week. From what I can see from the [Angular 18 Changelog](https://github.com/angular/angular/blob/main/CHANGELOG.md#1800-2024-05-22) there isn't anything for me to believe the changes to `@angular/core` and other packages cause our test harness to break. With that being said, the only requirement was to add support in the scaffolded dependencies and added Angular 18 system tests.

This PR also updates the angular 17 system test to not use the release candidate and instead use the published version

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Comment by AtofStryker** [comment]
This PR does not support angular signals, but that is being worked on separately. We likely want to introduce this PR first so users can get scaffolded angular 18 support in the CLI, then introduce our new signals test harness that users consuming signals in angular 17 and 18 can leverage. At some point, we will want to merge this new test harness upstream and deprecate support for older versions of angular

File: cli/CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,20 @@
...
+
+_Released 6/18/2024 (PENDING)_
+
+**Features:**
+
+- Adds Component Testing support for Angular version 18. Addresses [#29309](https://github.com/cypress-io/cypress/issues/29309).
+
+**Bugfixes:**
+
+- Fixed an issue where `inlineSourceMaps` was still being used when `sourceMaps` was provided in a users typescript config for typescript version 5. Fixes [#26203](https://github.com/cypress-io/cypress/issues/26203).
```
Comment (jordanpowell88) [inline cli/CHANGELOG.md:12]: This is a wonderful fix BTW. I run into this all the time when running Angular apps with CT 👍

File: system-tests/projects/angular-17/package.json [hunk 2]
```diff
@@ -3,29 +3,29 @@
...
   "dependencies": {
-    "@angular/animations": "^17.0.0-rc.2",
-    "@angular/common": "^17.0.0-rc.2",
-    "@angular/compiler": "^17.0.0-rc.2",
-    "@angular/core": "^17.0.0-rc.2",
-    "@angular/forms": "^17.0.0-rc.2",
-    "@angular/platform-browser": "^17.0.0-rc.2",
-    "@angular/platform-browser-dynamic": "^17.0.0-rc.2",
-    "@angular/router": "^17.0.0-rc.2",
+    "@angular/animations": "^17.3.10",
```
Comment (jordanpowell88) [inline system-tests/projects/angular-17/package.json:13]: Great job cleaning this up btw.

**Review by jordanpowell88 — APPROVED** [review]
PR looks great @AtofStryker. Thank you for picking this up

File: cli/CHANGELOG.md (hunk 1 above)
Comment (AtofStryker) [reply cli/CHANGELOG.md:12]: Thank you. It actually went in with #29568 but I forgot to update the changelog after release so the change entry was in the wrong cypress version 😬 .

**Review by cacieprins — APPROVED** [review]

**Pull request merged**